### PR TITLE
Typescript conversion for manifest-parser (PROPOSED)

### PR DESCRIPTION
**Pull request by lindner**
- introduce ts-pegjs
- lint fixes
  - switch from var to let/const
  - consistent return from functions
  - triple-equals and more.
- use exported parse() function
- move output to ts/manifest-parser.ts and git ignore it.
- add some explicit types in tools/sigh.js

### CODE REVIEW

**Comment by lindner** [comment]
Looking for feedback on this approach.  I think it's worth it to do this; but could be swayed otherwise.

**Comment by shans** [comment]
Looks interesting! I'm going to throw discussion over to the chat for now, and we can make a decision in the weekly meeting.

File: tools/sigh.js [hunk 1]
```diff
@@ -154,23 +154,52 @@ function clean() {
...
+    plugins: [tsPegjsPlugin],
     output: 'source',
-    trace: false
+    trace: false,
+    tspegjs: {
+      noTsLint: false,
+      tslintIgnores: 'no-any,only-arrow-functions,max-line-length,align,trailing-comma,interface-name,switch-default,object-literal-shorthand'
+    },
+    returnTypes: {
+      'Annotatiopn': 'string',
```
Comment (piotrswigon) [inline tools/sigh.js]: Typo.

**Comment by lindner** [comment]
Going to follow up on https://github.com/pegjs/pegjs/issues/562

Pushing the types into PEG directly would make this all a lot less brittle.  As it stands I'd want to write some tests to verify that the generated code was typed sufficiently.

**Comment by lindner** [comment]
Thanks to @Cypher1 for taking this to conclusion!

**Pull request closed without merging**